encoding/marshal_test.go

```diff
@@ -216,18 +216,58 @@ func TestEncodeDecode(t *testing.T) {
 	}
 }
 
+func TestEncodeAll(t *testing.T) {
+	var expected []byte
+	for i := range testStructs {
+		expected = append(expected, Marshal(testStructs[i])...)
+	}
+
+	b := new(bytes.Buffer)
+	NewEncoder(b).EncodeAll(testStructs...)
+	if !bytes.Equal(b.Bytes(), expected) {
+		t.Errorf("expected %v, got %v", expected, b.Bytes())
+	}
+}
+
+func TestDecodeAll(t *testing.T) {
+	b := new(bytes.Buffer)
+	enc := NewEncoder(b)
+	for i := range testStructs {
+		enc.Encode(testStructs[i])
+	}
+
+	var emptyStructs = []interface{}{&test0{}, &test1{}, &test2{}, &test3{}, &test4{}, &test5{}, &test6{}}
+	err := NewDecoder(b).DecodeAll(emptyStructs...)
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
This check is not good enough, you definitely need to write some comparisons to make sure the data was populated correctly. At the very least you should be checking correctness for test0 and test6 (if it's getting things wrong, it's most likely to be getting them wrong on the edges), but really you should be checking all of them.